src/test/java/edu/harvard/iq/dataverse/export/DDIExporterTest.java

```diff
@@ -42,7 +47,10 @@
     private static final SettingsServiceBean settingsService = Mockito.mock(SettingsServiceBean.class);
     private static final LicenseServiceBean licenseService = Mockito.mock(LicenseServiceBean.class);
     private static final MockDatasetFieldSvc datasetFieldTypeSvc = new MockDatasetFieldSvc();
-    private static final Gson gson = new Gson();
+    private static final Gson gson = new GsonBuilder().registerTypeAdapter(LocalDate.class, (JsonDeserializer<LocalDateTime>) (JsonElement json, Type type, JsonDeserializationContext jsonDeserializationContext) -> {
+        Instant instant = Instant.ofEpochMilli(json.getAsJsonPrimitive().getAsLong());
+        return LocalDateTime.ofInstant(instant, ZoneId.systemDefault());
+    }).create();
```

Comment:
I ran this test on my laptop and it works fine. It seems like a lot of boilerplate code.

I noticed that the `InaccessibleObjectException` error was first reported by @poikilotherm in an issue about Java 17:

- https://github.com/IQSS/dataverse/issues/8094#issuecomment-1022101843

@ErykKul are you running something newer than Java 11? In that issue, @poikilotherm suggests at upgrading Gson might fix this and what is perhaps a related issued here:

- https://github.com/google/gson/issues/1875

I can't reproduce the problem on Java 11 and haven't tried on anything newer. We aren't actively working on that Java 17 issue above but this pull request seems like a step in the right direction. I'm wondering, however, can you please try upgrading Gson to see if that's also a suitable fix?

Reply:
I have tried upgrading gson to 2.9.0, but it did not solve the problem. Downgrading to jdk11 helped, the test passes and in stead of errors I am getting warnings (with both gson versions; 2.9.0 and 2.2.4):
![image](https://user-images.githubusercontent.com/101262459/159464268-285c6a6c-cb85-41bb-b0e9-538596a94e30.png)